transcendence-app/src/chat/chatroom/chatroom-member/chatroom-member.service.ts

```diff
@@ -38,6 +38,26 @@ export class ChatroomMemberService {
     return chatMember && !chatMember.banned ? chatMember : null;
   }
 
+  async leaveChatroom(
+    chatroomId: string,
+    userId: string,
+  ): Promise<ChatroomMember | null> {
+    const foundChatroomMember = await this.chatroomMemberRepository.getById(
+      chatroomId,
+      userId,
+    );
+
+    if (!foundChatroomMember) {
+      throw new NotFoundException();
+    }
+    if (foundChatroomMember.banned || foundChatroomMember.muted) {
+      return this.chatroomMemberRepository.updateById(chatroomId, userId, {
+        joinedAt: null,
```

Comment:
Sorry but I don't see it, why updating it with `joinedAt: null` and not deleting it?

Reply:
[Here](https://42born2code.slack.com/archives/C033YQ28QA0/p1666802987816889?thread_ts=1666774111.696829&cid=C033YQ28QA0)

Reply:
Undestood, thanks

Reply:
:+1: